src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -194,7 +193,10 @@ public void actionPerformed(@NotNull AnActionEvent e) {
     executor.execute(() -> {
       try {
         autoInstallDone.get();
-        if (projectSettingsImported.get() && importIdeSettings && ideSettingsImported.get()) {
+        var settingsPresent = projectSettingsImported.get()
```

Comment:
`var` 😝

Reply:
Good reason to migrate to Java 11 :tada:

Reply:
I assume sonar complains that `Boolean` can be null? In that case the new version `projectSettingsImported.get() && importIdeSettings` would still throw unfortunately

Reply:
yeah, and I just bluntly coded their suggestion, please, feel free to change properly

Reply:
there'll be more PRs on this topic soonish, so you can plug in and fix this at any time later too @nikke234

Reply:
Yeah will do at some point :+1: